File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_think/20260419_114743/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"time"
)

type Relative struct {
	FirstName    string `json:"FirstName"`
	LastName     string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

type Person struct {
	FirstName string     `json:"FirstName"`
	LastName  string     `json:"LastName"`
	Birthday  string     `json:"Birthday"`
	Age       int        `json:"Age"`
	Relatives []Relative `json:"Relatives"`
}

func main() {
	file, err := os.Open("input/input.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	reader.FieldsPerRecord = -1
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	var people []Person
	refDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)

	for i, record := range records {
		if i == 0 {
			continue
		}
		if len(record) < 2 {
			continue
		}

		name := strings.TrimSpace(record[0])
		birthdayStr := strings.TrimSpace(record[1])

		nameParts := strings.Fields(name)
		if len(nameParts) < 2 {
			continue
		}
		firstName := nameParts[0]
		lastName := nameParts[len(nameParts)-1]

		birthDate, err := time.Parse("1/2/2006", birthdayStr)
		if err != nil {
			continue
		}

		age := refDate.Year() - birthDate.Year()
		if refDate.Month() < birthDate.Month() || (refDate.Month() == birthDate.Month() && refDate.Day() < birthDate.Day()) {
			age--
		}

		var relatives []Relative
		relCols := []string{"Father", "Mother", "Brother", "Sister"}
		for j, col := range relCols {
			if j+2 < len(record) {
				val := strings.TrimSpace(record[j+2])
				if val != "null" && val != "" {
					parts := strings.Fields(val)
					if len(parts) >= 2 {
						relatives = append(relatives, Relative{
							FirstName:    parts[0],
							LastName:     parts[len(parts)-1],
							Relationship: col,
						})
					}
				}
			}
		}

		people = append(people, Person{
			FirstName: firstName,
			LastName:  lastName,
			Birthday:  birthDate.Format("2006-01-02"),
			Age:       age,
			Relatives: relatives,
		})
	}

	jsonData, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(string(jsonData))
}

File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_think/20260419_114743/python/007_beatles_interview/solution.py
import csv
import json
from datetime import datetime, date

def main():
    ref_date = date(2025, 7, 1)
    result = []
    
    with open('input/input.csv', 'r') as f:
        reader = csv.DictReader(f)
        for row in reader:
            name_parts = row['Name'].strip().split()
            first_name = name_parts[0]
            last_name = name_parts[-1]
            
            bday_date = datetime.strptime(row['Birthday'].strip(), "%m/%d/%Y").date()
            bday_str = bday_date.strftime('%Y-%m-%d')
            
            age = ref_date.year - bday_date.year
            if (ref_date.month, ref_date.day) < (bday_date.month, bday_date.day):
                age -= 1
                
            relatives = []
            for rel_type in ['Father', 'Mother', 'Brother', 'Sister']:
                rel_name = row[rel_type].strip()
                if rel_name.lower() != 'null':
                    rel_parts = rel_name.split()
                    relatives.append({
                        "FirstName": rel_parts[0],
                        "LastName": rel_parts[-1],
                        "Relationship": rel_type
                    })
                    
            result.append({
                "FirstName": first_name,
                "LastName": last_name,
                "Birthday": bday_str,
                "Age": age,
                "Relatives": relatives
            })
            
    print(json.dumps(result, indent=2))

if __name__ == '__main__':
    main()